include/crow/middlewares/cookie_parser.h

```diff
@@ -30,10 +32,131 @@ namespace crow
 
     struct CookieParser
     {
+        // Cookie stores key, value and attributes
+        struct Cookie
+        {
+            enum class SameSitePolicy
+            {
+                Strict,
+                Lax,
+                None
+            };
+
+            template<typename U>
+            Cookie(const std::string& key, U&& value):
+              Cookie()
+            {
+                key_ = key;
+                value_ = std::forward<U>(value);
+            }
+
+            // format cookie to HTTP header format
+            std::string format() const
+            {
+                const static std::string DIVIDER = "; ";
+                const static char* HTTP_DATE_FORMAT = "%a, %d %b %Y %H:%M:%S GMT";
+
+                std::stringstream ss;
+                ss << key_ << '=';
+                ss << (value_.empty() ? "\"\"" : value_);
+                if (expires_at_)
+                {
+                    ss << DIVIDER << "Expires=";
+                    ss << std::put_time(expires_at_.get_ptr(), HTTP_DATE_FORMAT);
+                }
+                if (max_age_) ss << DIVIDER << "Max-Age=" << *max_age_;
+                if (!domain_.empty()) ss << DIVIDER << "Domain=" << domain_;
+                if (!path_.empty()) ss << DIVIDER << "Path=" << path_;
+                if (secure_) ss << DIVIDER << "Secure";
+                if (httponly_) ss << DIVIDER << "HttpOnly";
+                if (same_site_)
```

Comment:
I wonder if there's a better way to handle this (I really don't like having many if statements right after each other).
If there isn't then I guess it's not that bad.

Reply:
I've moved some of the to dumpString, this should be more readable now